File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/CreateEventMutationResponse.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

/**
 * Create a new event.
 */
open class CreateEventMutationResponse : GraphQLResult<MutableMap<String, Event>>() {

    companion object {
        const val OPERATION_NAME: String = "createEvent"
    }

    /**
     * Create a new event.
     */
    fun createEvent(): Event {
        val data: MutableMap<String, Event> = super.getData()
        return data.getValue(OPERATION_NAME)
    }
}

File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/CreateEventMutationResponse.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

/**
 * Create a new event.
 */
class CreateEventMutationResponse extends GraphQLResult[JMap[String, Event]] {

    /**
     * Create a new event.
     */
    def createEvent(): Event = {
        val data: JMap[String, Event] = getData
        if (data != null) data.get(CreateEventMutationResponse.OPERATION_NAME) else null
    }

}

object CreateEventMutationResponse {

    private final val OPERATION_NAME: String = "createEvent"

}
